examples/render_grid/run.py

```diff
@@ -0,0 +1,334 @@
+                final_tone_curve_hdr_component_shared_memory_handles
+            )
+
+    with instance.end_frame():
+        pass
+
+    for frame_idx in range(600):
+        with instance.begin_frame():
+            # move the parent actor around
+            location = view_target_pov["location"]
+            t = frame_idx / 30 * 2 * np.pi
+
+            # walk foward along the lookat direction, but don't change the height above the floor
+            x = frame_idx
+            # Add a figure 8 pattern in the camera plane
+            y = np.sin(t)
+            z = np.sin(t) * np.cos(t)
+
+            new_location = {
+                "x": float(
+                    location["x"]
+                    + actor_forward_vector[0] * x
+                    + actor_right_vector[0] * 20 * y
+                    + actor_up_vector[0] * 20 * z
+                ),
+                "y": float(
+                    location["y"]
+                    + actor_forward_vector[1] * x
+                    + actor_right_vector[1] * 20 * y
+                    + actor_up_vector[1] * 20 * z
+                ),
+                "z": float(
+                    location["z"]
+                    # + actor_forward_vector[2] * x # don't walk down into the floor
+                    + actor_right_vector[2] * 20 * y
+                    + actor_up_vector[2] * 20 * z
+                ),
+            }
+
+            # move the parent actor
+            instance.unreal_service.call_function(
+                uobject=parent_actor,
+                ufunction=AActor_K2_SetActorLocation,
+                args={"NewLocation": new_location},
+            )
+
+        with instance.end_frame():
+            # build a montage of the rendered frames
+            viewport_y = int(viewport_y)
+            viewport_x = int(viewport_x)
+            canvas = np.zeros((viewport_y * rows, viewport_x * cols, 4), dtype=np.uint8)
+            for idx, (
+                final_tone_curve_hdr_component_shared_memory_handles,
+                final_tone_curve_hdr_component,
+            ) in enumerate(
+                zip(
+                    final_tone_curve_hdr_component_shared_memory_handles_list,
+                    final_tone_curve_hdr_components,
+                )
+            ):
+                i = idx // cols
+                j = idx % cols
+                return_values = instance.sp_func_service.call_function(
+                    uobject=final_tone_curve_hdr_component,
+                    function_name="read_pixels",
+                    uobject_shared_memory_handles=final_tone_curve_hdr_component_shared_memory_handles,
+                )
+                data = return_values["arrays"]["data"]
+                canvas[
+                    i * viewport_y : (i + 1) * viewport_y,
+                    j * viewport_x : (j + 1) * viewport_x,
+                ] = data
+
+        cv2.imwrite(f"montage_{frame_idx}.png", canvas)
+        print(f"Saved montage_{frame_idx}.png")
```

Comment:
For consistency with other examples:

```
spear.log(f"Saved montage_{frame_idx}.png")
```

(Also please add logic to save with a consistent number of leading zeros.)